features/endpoints/grafana/client/src/main/java/org/opennms/netmgt/endpoints/grafana/client/GrafanaClientImpl.java

```diff
@@ -122,12 +126,12 @@ public Dashboard getDashboardByUid(String uid) throws IOException {
     }
 
     @Override
-    public byte[] renderPngForPanel(Dashboard dashboard, Panel panel, int width, int height, long from, long to, Map<String, String> variables) throws IOException {
+    public CompletableFuture<byte[]> renderPngForPanel(Dashboard dashboard, Panel panel, int width, int height, long from, long to, Map<String, String> variables) {
         final HttpUrl.Builder builder = baseUrl.newBuilder()
                 .addPathSegment("render")
                 .addPathSegment("d-solo")
                 .addPathSegment(dashboard.getUid())
-                .addPathSegments("flow"); //FIXME: What should this be?
+                .addPathSegments("z"); // We need some string here, but it doesn't seem to matter what it is
```

Comment:
I tried removing this, but still appears to be necessary.

Reply:
Yeah you are right, I remember getting this to work, but probably were trying to get the dashboard vs rendering it.

Generally it seems that the official way to render is by its `slug` instead of the UID. I guess we are hoping for this to never break? (-:

Reply:
We'll eventually have to account for API incompatibilities between Grafana versions.

Did you manage to find a reference to the expected URL format to render?

Reply:
I only found this one: https://grafana.com/docs/reference/sharing/ -> Direct Link Rendered Image.

They use the following URL-Schema from what I can tell: `http://grafana:3600/render/d-solo/db/<dashboard-slug>?panelId=<id>&from=<from>&to=<to>&width=<width>&height=<height>`